### DISCUSSION TITLE
Do we need to perform server backups?

### DISCUSSION THREAD

**Discussion by briandipalma** [immich-app/immich#3299] [Q&A] [answered]
If I have a backup of the upload location and I setup a new Immich instance exactly the same a previous instance what do I need a DB backup for? Can I not just trigger metadata and thumbnail jobs for the database to be recreated again?

I guess a backup would save albums? Can I not tag images as "Archive" to filter them out so I could recreate date based albums quickly?

**Comment by bo0tzz** [comment, marked as answer]
Immich saves file paths in the database. If you give it only a library folder, it won't know anything about the files in there. Not to mention albums, named people, archived or favourited assets, etc.

**Reply by briandipalma** [reply]
There is no job that makes Immich rescan the upload location again to make it add files that are missing from its database?

I understand losing albums etc but if Immich can't be made spot there is a difference between the upload location file structure and the DB and then reconciling those differences then a backup needs to be taken after every single change to the DB to be of any use.

Even a daily backup won't help if it misses 200-300 new images uploaded during an event for example. I'd have to manually upload the files from the *upload location* that have been added since the last backup. An odd situation/workflow I'm sure you'd agree?

**Reply by bo0tzz** [reply]
That is indeed how Immich works, as it was designed upload-first. 

> Even a daily backup won't help if it misses 200-300 new images uploaded during an event for example.

That applies to any backups, unrelated to how Immich handles files.

**Reply by briandipalma** [reply]
OK, that's fine, I just wanted to make sure I understand the limitations of backups. Would there be an objection to creating a feature request discussion for reconciling the library - although it seems adjacent to the file watching PR that's already been worked on so maybe once that's in this feature would be easier to implement.

**Reply by maltokyo** [reply]
This is fixed / added now with Libraries functionality, correct?

**Reply by briandipalma** [reply]
I thought Libraries were external to Immich? It wouldn't upload images to them and wouldn't rename files in them based on naming configuration?

**Comment by doswear-wong** [comment]
After fresh new installation, which only copy the library folder, resulted in same consequence. what can be done to solve the problem now

**Reply by briandipalma** [reply]
You need to re-upload all your files into Immich using the CLI. Bit of a drag I'm afraid.

**Reply by coltenkrauter** [reply]
This caught me off guard... I recently got Immich setup, I hadn't yet setup backups, but I had the library storage directory backedup. I had an issue on TrueNas Scale where Immich stopped working (not an issue with Immich itself) and I had to re-setup immich. 

I was expecting that if I mounted the library to the new instance, it would scan it and get working.... But it appears that isn't the case and I will have to re-upload everything. Over 5 TB... Bummer. 😬

**Reply by coltenkrauter** [reply]
Will re-uploading restore all my albums and metadata?

**Reply by alextran1502** [reply]
@coltenkrauter That was a comment a year ago... You can use the library feature https://immich.app/docs/features/libraries

**Reply by briandipalma** [reply]
Sadly Immich doesn't save data like albums to the file metadata so you'll have to recreate them. It's why I won't use Immich albums and would like support for tags in Immich which can be saved to the file. Albums are redundant once you have good tag metadata support.

**Reply by EdeAbreu23** [reply]
> Sadly Immich doesn't save data like albums to the file metadata so you'll have to recreate them. It's why I won't use Immich albums and would like support for tags in Immich which can be saved to the file. Albums are redundant once you have good tag metadata support.

so if i create albums on immich and for some reason my server die, and i have to restore everything or lets say i want to switch from immich to another software, those albums will just be lost?

**Reply by briandipalma** [reply]
@elvis0288 Yep all gone.

**Reply by EdeAbreu23** [reply]
That sucks so what would you suggest I do then? Wait for the tags?

**Reply by briandipalma** [reply]
Backing up wouldn't be required if all the user's modifications were written to the image metadata. That's how I'd architect self hosted software. Everything built from media metadata and text files for configuration. I think the way the *arr programs and Immich were designed was influenced by enterprise work practices. Grand in my day job as we've teams of people to think about the backing up process. Not so when it's a self hosted project for people with limited time or skill (not all users are devs or dev-ops). Although I think it's good practice to be able to scrap the entire stack and have it back up and running with the exact same state as long as you backed up the actual media.

**Comment by maltokyo** [comment]
Just back up?
